Civi/Funding/FundingCase/Actions/DefaultFundingCaseActionsDeterminer.php

```diff
@@ -80,4 +97,26 @@ private function isApprovePossible(array $applicationProcessStatusList): bool {
     return $eligibleCount > 0;
   }
 
+  /**
+   * @phpstan-param array<int, \Civi\Funding\Entity\FullApplicationProcessStatus> $applicationProcessStatusList
+   */
+  private function isFinishClearingPossible(array $applicationProcessStatusList): bool {
+    foreach ($applicationProcessStatusList as $applicationProcessId => $applicationProcessStatus) {
+      // Eligibility of all applications has to be decided.
+      if (NULL === $this->statusInfo->isEligibleStatus($applicationProcessStatus->getStatus())) {
+        return FALSE;
+      }
+
+      if (TRUE === $this->statusInfo->isEligibleStatus($applicationProcessStatus->getStatus())) {
+        // There has to be a clearing process for every eligible application that is either accepted or rejected.
+        $clearingProcess = $this->clearingProcessManager->getByApplicationProcessId($applicationProcessId);
+        if (NULL === $clearingProcess || !in_array($clearingProcess->getStatus(), ['accepted', 'rejected'], TRUE)) {
+          return FALSE;
+        }
+      }
```

Comment:
how to handle `FALSE === $this->statusInfo->isEligibleStatus($applicationProcessStatus->getStatus()` ?
if every call to `isEligibleStatus()` in this `foreach` loop returns `FALSE`, then `isFinishClearingPossible()` would return `TRUE`

Reply:
If the funding case contains application processes of which the eligibility hasn't been decided, that's completely fine. However, for every eligible application process there has to be a clearing process which is either accepted or rejected.